ninja-maven-plugin/src/main/java/ninja/maven/NinjaRunMojo.java

```diff
@@ -297,20 +358,31 @@ else if (file.getName().endsWith(".jar") && this.watchAllClassPathJars) {
         
         jvmArguments.add(systemPropertyDevMode);
 
-        String portSelection
-                = "-D" + NinjaJetty.COMMAND_LINE_PARAMETER_NINJA_PORT + "=" + port;
-
-        jvmArguments.add(portSelection);
+        if (port != null) {
+            String portSelection
+                    = "-D" + NinjaJetty.KEY_NINJA_PORT + "=" + port;
+            jvmArguments.add(portSelection);
+        }
         
-        if (contextPath != null) {
-            String systemPropertyContextPath = "-Dninja.context=" + contextPath;
+        if (getContextPath() != null) {
+            String systemPropertyContextPath = "-Dninja.context=" + getContextPath();
             jvmArguments.add(systemPropertyContextPath);
         }
         
+        if (jvmArgs != null) {
+            // use excellent DrJava library to tokenize arguments (do not keep tokens)
+            List<String> tokenizedArgs = ArgumentTokenizer.tokenize(jvmArgs, false);
```

Comment:
Hmm.... Why don't we just use Guava's splitter to split the stuff easily? Copying the DrJava file with only use case... hmm...